src/core/service.c

```diff
@@ -2631,14 +2665,9 @@ static void service_run_next_control(Service *s) {
         pidref_done(&s->control_pid);
 
         r = service_spawn(s,
+                          s->control_command_id,
                           s->control_command,
                           timeout,
-                          EXEC_APPLY_SANDBOXING|EXEC_APPLY_CHROOT|EXEC_IS_CONTROL|
-                          (IN_SET(s->state, SERVICE_CONDITION, SERVICE_START_PRE, SERVICE_START, SERVICE_START_POST, SERVICE_RUNNING, SERVICE_RELOAD) ? EXEC_WRITE_CREDENTIALS : 0)|
```

Comment:
Here, EXEC_WRITE_CREDENTIALS is set when the state is `SERVICE_CONDITION`. So, I think we should enable credentials for commands `SERVICE_EXEC_CONDITION`, no?

Reply:
As mentioned in commit msg, I think this is accidentally introduced in https://github.com/systemd/systemd/pull/27279. I don't think this is desirable, and the first `ExecCondition=` cannot access creds anyway. So the incompatibility introduced should be negligible.

Reply:
OK, but the commit message mention ExecReload. Please also mention ExecCondition somewhere.

Reply:
Oh, in the revised commit, it is mentioned. Thank you.